From 3a0332a2cbf70f84440eeb249441f9369ad0e23d Mon Sep 17 00:00:00 2001
From: Tomer Maimon <tmaimon77@gmail.com>
Date: Wed, 10 Apr 2024 10:53:49 +0300
Subject: [PATCH 06/23] arm64: dts: nuvoton: add npcm8xx nodes

Add the follwoing nodes to NPCM8xx dts:
- fuse
- sp-ctrl
- ohci
- PCI mailbox
- tip mailbox
- jtag master
- eSPI virtual wire GPIO
- tmps
- sgpio
- thermal-zone
- smb6, smb7, bu4-bu6 pins
- optee
- mc

Signed-off-by: Tomer Maimon <tmaimon77@gmail.com>
---
 .../dts/nuvoton/nuvoton-common-npcm8xx.dtsi   | 238 ++++++++++++++++--
 .../boot/dts/nuvoton/nuvoton-npcm845.dtsi     |  33 ++-
 2 files changed, 247 insertions(+), 24 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
index 72e6859e8b1d..92717af25e18 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
@@ -34,6 +34,12 @@ rst: system-controller@f0801000 {
 			reg = <0x0 0xf0801000 0x0 0xC4>;
 		};
 
+		fuse:fuse@f0189000 {
+			compatible = "nuvoton,npcm845-fuse", "syscon",
+				"simple-mfd";
+			reg = <0x0 0xf0189000 0x0 0x1000>;
+		};
+
 		gic: interrupt-controller@dfff9000 {
 			compatible = "arm,gic-400";
 			reg = <0x0 0xdfff9000 0x0 0x1000>,
@@ -50,6 +56,12 @@ ppi_cluster0: interrupt-partition-0 {
 				};
 			};
 		};
+
+		spswc: spswc@f0800038 {
+			compatible = "nuvoton,npcm-sp-ctrl";
+			reg = <0x0 0xf0800038 0x0 0x4>;
+		};
+
 	};
 
 	ahb {
@@ -146,6 +158,20 @@ ehci2: usb@f082a100 {
 			status = "disabled";
 		};
 
+		ohci1: usb@f0829000 {
+			compatible = "nuvoton,npcm750-ohci";
+			reg = <0x0 0xf0829000 0x0 0x1000>;
+			interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>;
+			status = "disabled";
+		};
+
+		ohci2: usb@f082b000 {
+			compatible = "nuvoton,npcm750-ohci";
+			reg = <0x0 0xf082b000 0x0 0x1000>;
+			interrupts = <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>;
+			status = "disabled";
+		};
+
 		sdhci: mmc@f0842000 {
 			compatible = "nuvoton,npcm845-sdhci";
 			reg = <0x0 0xf0842000 0x0 0x100>;
@@ -158,6 +184,24 @@ sdhci: mmc@f0842000 {
 			status = "disabled";
 		};
 
+		pcie: pcie@e1000000 {
+			compatible = "nuvoton,npcm845-pcie";
+			device_type = "pci";
+			reg = <0x0 0xE1000000 0x0 0x1000>,
+				<0x0 0xE8000000 0x0 0x1000>;
+			bus-range = <0x0 0xF>;
+			#address-cells = <3>;
+			#size-cells = <2>;
+			#interrupt-cells = <1>;
+			ranges = <0x01000000 0 0xe9000000 0x0 0xe9000000 0 0x01000000
+				0x02000000 0 0xea000000 0x0 0xea000000 0 0x04000000>;
+			resets = <&rstc 0x34 15>;
+			interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map-mask = <0 0 0 7>;
+			interrupt-map = <0 0 0 1 &gic GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
+			nuvoton,sysgcr = <&gcr>;
+		};
+
 		fiu0: spi@fb000000 {
 			compatible = "nuvoton,npcm845-fiu";
 			#address-cells = <1>;
@@ -211,10 +255,11 @@ fiux: spi@fb001000 {
 		};
 
 		pcimbox: pcimbox@f0848000 {
-			compatible = "nuvoton,npcm845-pci-mbox";
+			compatible = "nuvoton,npcm845-pci-mbox", "nuvoton,npcm750-pci-mbox",
+					"simple-mfd", "syscon";
 			reg = <0x0 0xf084C000 0x0 0x8
 				0x0 0xf0848000 0x0 0x3F00>;
-			interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		vcd: vcd@f0810000 {
@@ -236,6 +281,14 @@ ece: video-codec@f0820000 {
 			status = "disabled";
 		};
 
+		tip_mbox: mailbox@f080d000 {
+			compatible = "nuvoton,npcm845-mbox";
+			reg = <0 0xf080d000 0 0x1000>;
+			interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
+			#mbox-cells = <2>;
+			status = "disabled";
+		};
+
 		apb {
 			#address-cells = <1>;
 			#size-cells = <1>;
@@ -280,14 +333,46 @@ kcs3: kcs3@0 {
 				};
 			};
 
-			bpc: bpc@7040 {
-				compatible = "nuvoton,npcm845-bpc";
-				reg = <0x7040 0x20>;
-				interrupts = <GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>;
+			lpc_host: lpc_host@7000 {
+				compatible = "nuvoton,npcm845-lpc-host",
+						"simple-mfd", "syscon";
+				reg = <0x7000 0x60>;
+
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges = <0x0 0x7000 0x60>;
+
+				lpc_bpc: lpc_bpc@40 {
+					compatible = "nuvoton,npcm845-lpc-bpc", "nuvoton,npcm750-lpc-bpc";
+					reg = <0x40 0x20>;
+					interrupts = <GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>;
+					status = "disabled";
+				};
+			};
+
+			espi: espi@9f000 {
+				compatible = "nuvoton,npcm845-espi",
+						"simple-mfd", "syscon";
+				reg = <0x9f000 0x1000>;
+				vw_gpio: vw_gpio {
+					compatible = "nuvoton,npcm845-espi-vwgpio";
+					interrupts = <GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH>;
+					gpio-controller;
+					#gpio-cells = <2>;
+					status = "disabled";
+				};
+			};
+
+			peci0: peci-controller@100000 {
+				compatible = "nuvoton,npcm845-peci";
+				reg = <0x100000 0x1000>;
+				interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clk NPCM8XX_CLK_APB3>;
+				cmd-timeout-ms = <1000>;
 				status = "disabled";
 			};
 
-			spi1: spi@201000 {
+			pspi: spi@201000 {
 				compatible = "nuvoton,npcm845-pspi";
 				reg = <0x201000 0x1000>;
 				pinctrl-names = "default";
@@ -301,12 +386,31 @@ spi1: spi@201000 {
 				status = "disabled";
 			};
 
-			peci0: peci-controller@100000 {
-				compatible = "nuvoton,npcm845-peci";
-				reg = <0x100000 0x1000>;
-				interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&clk NPCM8XX_CLK_APB3>;
-				cmd-timeout-ms = <1000>;
+			jtm1: jtm@208000 {
+				compatible = "nuvoton,npcm845-jtm";
+				reg = <0x208000 0x1000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&jm1_pins>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				interrupts = <GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clk NPCM8XX_CLK_APB5>;
+				clock-names = "clk_apb5";
+				resets = <&rstc 0x74 29>;
+				status = "disabled";
+			};
+
+			jtm2: jtm@209000 {
+				compatible = "nuvoton,npcm845-jtm";
+				reg = <0x209000 0x1000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&jm2_pins>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				interrupts = <GIC_SPI 189 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clk NPCM8XX_CLK_APB5>;
+				clock-names = "clk_apb5";
+				resets = <&rstc 0x74 30>;
 				status = "disabled";
 			};
 
@@ -360,6 +464,8 @@ serial4: serial@4000 {
 				clocks = <&clk NPCM8XX_CLK_UART2>;
 				interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&bu4_pins>;
 				status = "disabled";
 			};
 
@@ -369,6 +475,8 @@ serial5: serial@5000 {
 				clocks = <&clk NPCM8XX_CLK_UART2>;
 				interrupts = <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&bu5_pins>;
 				status = "disabled";
 			};
 
@@ -378,6 +486,8 @@ serial6: serial@6000 {
 				clocks = <&clk NPCM8XX_CLK_UART2>;
 				interrupts = <GIC_SPI 198 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&bu6_pins>;
 				status = "disabled";
 			};
 
@@ -392,8 +502,9 @@ adc: adc@c000 {
 				compatible = "nuvoton,npcm845-adc";
 				reg = <0xC000 0x8>;
 				interrupts = <GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&clk NPCM8XX_CLK_AHB>;
+				clocks = <&clk NPCM8XX_CLK_ADC>;
 				resets = <&rstc 0x20 27>;
+				syscon = <&fuse>;
 				status = "disabled";
 			};
 
@@ -424,6 +535,42 @@ watchdog2: watchdog@a01c {
 				syscon = <&gcr>;
 			};
 
+			tmps: tmps@188000 {
+				compatible = "nuvoton,npcm845-thermal";
+				interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clk NPCM8XX_CLK_APB4>;
+				reg = <0x188000 0x5C>;
+				#thermal-sensor-cells = <1>;
+			};
+
+			sgpio1: sgpio@101000 {
+				compatible = "nuvoton,npcm845-sgpio";
+				reg = <0x101000 0x200>;
+				clocks = <&clk NPCM8XX_CLK_APB3>;
+				interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&iox1_pins>;
+				nuvoton,input-ngpios = <64>;
+				nuvoton,output-ngpios = <64>;
+				status = "disabled";
+			};
+
+			sgpio2: sgpio@102000 {
+				compatible = "nuvoton,npcm845-sgpio";
+				reg = <0x102000 0x200>;
+				clocks = <&clk NPCM8XX_CLK_APB3>;
+				interrupts = <GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&iox2_pins>;
+				nuvoton,input-ngpios = <64>;
+				nuvoton,output-ngpios = <64>;
+				status = "disabled";
+			};
+
 			pwm_fan:pwm-fan-controller@103000 {
 				compatible = "nuvoton,npcm845-pwm-fan";
 				reg = <0x103000 0x3000>,
@@ -934,6 +1081,33 @@ gfxi: gfxi@e000 {
 		};
 	};
 
+	thermal-zones {
+		cpu0_thermal: cpu0-thermal {
+			polling-delay-passive = <250>;
+			polling-delay = <1000>;
+			thermal-sensors = <&tmps 0>;
+			trips {
+				cpu0_alert0: trip-point0 {
+					temperature = <90000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+			};
+		};
+		cpu1_thermal: cpu1-thermal {
+			polling-delay-passive = <250>;
+			polling-delay = <1000>;
+			thermal-sensors = <&tmps 1>;
+			trips {
+				cpu1_alert0: trip-point0 {
+					temperature = <90000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+			};
+		};
+	};
+
 	pinctrl: pinctrl@f0800260 {
 		compatible = "nuvoton,npcm845-pinctrl", "syscon", "simple-mfd";
 		ranges = <0x0 0x0 0xf0010000 0x8000>;
@@ -1538,10 +1712,34 @@ smb6_pins: smb6-pins {
 			groups = "smb6";
 			function = "smb6";
 		};
+		smb6b_pins: smb6b-pins {
+			groups = "smb6b";
+			function = "smb6b";
+		};
+		smb6c_pins: smb6c-pins {
+			groups = "smb6c";
+			function = "smb6c";
+		};
+		smb6d_pins: smb6d-pins {
+			groups = "smb6d";
+			function = "smb6d";
+		};
 		smb7_pins: smb7-pins {
 			groups = "smb7";
 			function = "smb7";
 		};
+		smb7b_pins: smb7b-pins {
+			groups = "smb7b";
+			function = "smb7b";
+		};
+		smb7c_pins: smb7c-pins {
+			groups = "smb7c";
+			function = "smb7c";
+		};
+		smb7d_pins: smb7d-pins {
+			groups = "smb7d";
+			function = "smb7d";
+		};
 		spi1_pins: spi1-pins {
 			groups = "spi1";
 			function = "spi1";
@@ -1662,6 +1860,18 @@ hgpio7_pins: hgpio7-pins {
 			groups = "hgpio7";
 			function = "hgpio7";
 		};
+		bu4_pins: bu4-pins {
+			groups = "bu4";
+			function = "bu4";
+		};
+		bu5_pins: bu5-pins {
+			groups = "bu5";
+			function = "bu5";
+		};
+		bu6_pins: bu6-pins {
+			groups = "bu6";
+			function = "bu6";
+		};
 		gpo187_pins: gpo187-pins {
 			groups = "gpo187";
 			function = "gpo187";
diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi
index 243635b2b419..7826410d46af 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi
@@ -74,13 +74,26 @@ timer {
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>;
 	};
 
+	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
+
 	udc0_phy: usb-phy {
 		#phy-cells = <0>;
 		compatible = "usb-nop-xceiv";
 	};
 
 	ahb {
-		udc0:usb@f0830000 {
+		mc: memory-controller@f0824000 {
+			compatible = "nuvoton,npcm845-memory-controller";
+			reg = <0x0 0xf0824000 0x0 0x1000>;
+			interrupts = <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
+		};
+
+		udc0:udc@f0830000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0830000 0x0 0x1000
 			       0x0 0xfffeb000 0x0 0x800>;
@@ -94,7 +107,7 @@ udc0:usb@f0830000 {
 			status = "disabled";
 		};
 
-		udc1:usb@f0831000 {
+		udc1:udc@f0831000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0831000 0x0 0x1000
 			       0x0 0xfffeb800 0x0 0x800>;
@@ -108,7 +121,7 @@ udc1:usb@f0831000 {
 			status = "disabled";
 		};
 
-		udc2:usb@f0832000 {
+		udc2:udc@f0832000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0832000 0x0 0x1000
 			       0x0 0xfffec000 0x0 0x800>;
@@ -122,7 +135,7 @@ udc2:usb@f0832000 {
 			status = "disabled";
 		};
 
-		udc3:usb@f0833000 {
+		udc3:udc@f0833000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0833000 0x0 0x1000
 			       0x0 0xfffec800 0x0 0x800>;
@@ -136,7 +149,7 @@ udc3:usb@f0833000 {
 			status = "disabled";
 		};
 
-		udc4:usb@f0834000 {
+		udc4:udc@f0834000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0834000 0x0 0x1000
 			       0x0 0xfffed000 0x0 0x800>;
@@ -150,7 +163,7 @@ udc4:usb@f0834000 {
 			status = "disabled";
 		};
 
-		udc5:usb@f0835000 {
+		udc5:udc@f0835000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0835000 0x0 0x1000
 			       0x0 0xfffed800 0x0 0x800>;
@@ -164,7 +177,7 @@ udc5:usb@f0835000 {
 			status = "disabled";
 		};
 
-		udc6:usb@f0836000 {
+		udc6:udc@f0836000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0836000 0x0 0x1000
 			       0x0 0xfffee000 0x0 0x800>;
@@ -178,7 +191,7 @@ udc6:usb@f0836000 {
 			status = "disabled";
 		};
 
-		udc7:usb@f0837000 {
+		udc7:udc@f0837000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0837000 0x0 0x1000
 			       0x0 0xfffee800 0x0 0x800>;
@@ -192,7 +205,7 @@ udc7:usb@f0837000 {
 			status = "disabled";
 		};
 
-		udc8:usb@f0838000 {
+		udc8:udc@f0838000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0838000 0x0 0x1000
 			       0x0 0xfffef000 0x0 0x800>;
@@ -207,7 +220,7 @@ udc8:usb@f0838000 {
 			status = "disabled";
 		};
 
-		udc9:usb@f0839000 {
+		udc9:udc@f0839000 {
 			compatible = "nuvoton,npcm845-udc";
 			reg = <0x0 0xf0839000 0x0 0x1000
 			       0x0 0xfffef800 0x0 0x800>;
-- 
2.43.0

